parsers/CA_SK.py

```diff
@@ -1,142 +1,108 @@
+import typing
 from datetime import datetime, timedelta
 from logging import Logger, getLogger
 
 from pytz import timezone
-from requests import Response, Session
+from requests import Session
 
+from electricitymap.contrib.config import ZoneKey
+from electricitymap.contrib.lib.models.event_lists import ProductionBreakdownList
+from electricitymap.contrib.lib.models.events import ProductionMix, TotalConsumption
 from parsers.lib.exceptions import ParserException
 
 TIMEZONE = timezone("America/Regina")
-
-# URLs for the different endpoints.
 PRODUCTION_URL = (
-    "https://www.saskpower.com/ignitionapi/PowerUseDashboard/GetPowerUseDashboardData"
+    "https://www.saskpower.com/ignitionapi/PowerUseDashboard"
+    "/GetPowerUseDashboardData"
 )
 CONSUMPTION_URL = "https://www.saskpower.com/ignitionapi/Content/GetNetLoad"
-
 PRODUCTION_MAPPING = {
     "Hydro": "hydro",
     "Wind": "wind",
     "Solar": "solar",
     "Natural Gas": "gas",
     "Coal": "coal",
-    "Other": "unknown",  # This is internal consumption, losses, heat recovery facilities and small independent power producers.
+    # "Other" represents internal consumption, losses, heat recovery facilities
+    # and small independent power producers.
+    "Other": "unknown",
 }
-
-USER_AGENT = "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.142 Safari/537.36"
-
-
-def validate_zone_key(zone_key: str) -> None:
-    if zone_key != "CA-SK":
-        raise ParserException(
-            "CA_SK.py",
-            f"CA_SK.py is not designed to parse zone_key: {zone_key}.",
-            zone_key,
-        )
+USER_AGENT = (
+    "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 "
+    "(KHTML, like Gecko) Chrome/75.0.3770.142 Safari/537.36"
+)
 
 
-def validate_no_datetime(target_datetime: datetime | None, zone_key) -> None:
+def _request(
+    session: Session | None,
+    target_datetime: datetime | None,
+    url: str,
+    zone_key: ZoneKey,
+) -> typing.Any:
```

Comment:
Since we are returning json the type should be `list | dict | None`, which makes the type checking a little bit better.

Reply:
I didn't want to restrict the return types because a JSON payload could feasibly be a `bool`, `float`, `int`, or `str` as well. In fact, while the production URL returns a `dict`, judging from the existing parser, the consumption source seems to return a `str` (it's hard to be sure while it's down). How about this?

```suggestion
) -> dict | str:
```